### DISCUSSION TITLE
Lesson 12, Part 6 - DepositCollateral function in decentralized stablecoin uint

### DISCUSSION THREAD

**Discussion by 0xwtfwtf** [Cyfrin/foundry-full-course-cu#768] [General]
Here is the depositCollateral function for the decentralized stablecoin project. My question is that I don't see a line in this code that actually says send the collateral from msg.sender's wallet to this contract...first we update the mapping (but does doing that force the collateral to be transferred?), then we emit an event, then we have a bool to check whether the transfer happened....but why is there no line that clearly sends the collateral?

```solidity
function depositCollateral(
        address tokenCollateralAddress, 
        uint256 amountCollateral) 
        external
        isAllowedCollateral(tokenCollateralAddress) 
        moreThanZero(amountCollateral) 
        nonReentrant
        {
            s_CollateralDeposited[msg.sender][tokenCollateralAddress] += amountCollateral;
            emit CollateralDeposited(msg.sender, tokenCollateralAddress, amountCollateral);
            bool success = IERC20(tokenCollateralAddress).transferFrom(msg.sender, address(this), amountCollateral);
            if (!success) {
                revert DSCEngine__TransferFailed()}
        }

**Comment by shikhar229169** [comment]
```solidity
bool success = IERC20(tokenCollateralAddress).transferFrom(msg.sender, address(this), amountCollateral);
```

This is the line. Before calling the depositCollateral function, user needs to approve the dscEngine contract with required amount of tokens to be deposited on the respective erc20 contract.
Then they can call the depositCollateral function, as they have approved the tokens, then the dscEngine can take out the tokens from the user's respective erc20 token balance.
